gateway/tests/api/test_v1_catalog.py

```diff
@@ -0,0 +1,131 @@
+"""Tests catalog APIs."""
+
+from django.contrib.auth import models
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from api.models import Program
+
+
+class TestCatalogApi(APITestCase):
+    """TestCatalogApi."""
+
+    fixtures = ["tests/fixtures/catalog_fixtures.json"]
+
+    def test_catalog_list_non_auth_user(self):
+        """Tests catalog list non-authenticated."""
+        url = reverse("v1:catalog-list")
+        response = self.client.get(url, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+
+        public_function = response.data[0]
+        self.assertEqual(public_function.get("available"), False)
+        self.assertEqual(public_function.get("title"), "Public-Function")
+        self.assertEqual(public_function.get("type"), Program.APPLICATION)
+
+    def test_catalog_list_with_auth_user_without_run_permission(self):
+        """Tests catalog list authenticated without run permission."""
+        user = models.User.objects.get(username="test_user")
+        self.client.force_authenticate(user=user)
+
+        url = reverse("v1:catalog-list")
+        response = self.client.get(url, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+
+        public_function = response.data[0]
+        self.assertEqual(public_function.get("available"), False)
+        self.assertEqual(public_function.get("title"), "Public-Function")
+        self.assertEqual(public_function.get("type"), Program.APPLICATION)
+
+    def test_catalog_list_with_auth_user_with_run_permission(self):
+        """Tests catalog list authenticated with run permission."""
+        user = models.User.objects.get(username="test_user_2")
+        self.client.force_authenticate(user=user)
+
+        url = reverse("v1:catalog-list")
+        response = self.client.get(url, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+
+        public_function = response.data[0]
+        self.assertEqual(public_function.get("available"), True)
+        self.assertEqual(public_function.get("title"), "Public-Function")
+        self.assertEqual(public_function.get("type"), Program.APPLICATION)
+
+    def test_catalog_retrieve_non_auth_user(self):
+        """Tests catalog retrieve non-authenticated."""
+        url = reverse(
+            "v1:catalog-detail", args=["1a7947f9-6ae8-4e3d-ac1e-e7d608deec82"]
```

Comment:
I'm not sure how the `reverse` works.   Is this testing `retrieve` function?

Reply:
It's an utility from django, usually used in tests to not write every time the entire url:

https://docs.djangoproject.com/en/5.0/ref/urlresolvers/#reverse

In this case `v1:catalog-detail` points to `retrieve` but for example to access to the `programs/get_by_title` the reverse would be: `v1:programs-get-by-title`.

I usually get this info from this page:

<img width="640" alt="Screenshot 2024-08-16 at 3 35 03 PM" src="https://github.com/user-attachments/assets/bacae55d-4fd2-4020-9c4e-75bced3c2915">

I don't know, for me it's easier to manage it and we were using before in other tests too: https://github.com/Qiskit/qiskit-serverless/blob/main/gateway/tests/api/test_v1_program.py#L22

Happy to hear feedback if you prefer not use it. Unify approaches for me it's better 👌

Reply:
I don't know how 

> v1:catalog-detail points to retrieve

Doesn't `v1:catalog-retrieve` work?

Reply:
> Doesn't v1:catalog-retrieve work?

Surprisingly not, I had the same doubt 😂 

`list`, `retrieve`, `update` are built in methods created by DRF ViewSets. I suppose that something below attaches the `retrieve` method from DRF to the `detail` name in Django. From what I saw was the only one method that has a _special_ name.

Reply:
Have you tired just `v1:catalog`?   I have no idea so why the `detail` work.  Would you add comments on the `Reverse` with the `catalog-detail`?

Reply:
> Have you tired just v1:catalog?

Basically it follows the pattern: `<namespace>:<name>` where `<name>` is the name that django gives to the end-point (you can check an example in the image that I just shared a bit above).

> Would you add comments on the Reverse with the catalog-detail?

Sure, I will 👍